### PR TITLE
stacer: init at 1.1.0

### LINKED ISSUE #188921: Stacer
**Project description**
Stacer is an open source system optimizer and application monitor that helps users to manage entire system with different aspects, its an all in one system utility.

**Metadata**

* homepage URL: https://oguzhaninan.github.io/Stacer-Web/
* source URL: https://github.com/oguzhaninan/Stacer
* license:  GPL-3.0 
* platforms: linux

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Resolves #188921.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/stacer/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, qtcharts
+, qttranslations
```
Comment (sikmir) [inline pkgs/tools/system/stacer/default.nix:6]:
```suggestion
, qttools
```

File: pkgs/tools/system/stacer/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "oguzhaninan";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0qndzzkbq6abapvwq202kva8j619jdn9977sbqmmfs9zkjz4mbsd";
+  };
+
+  buildInputs = [
+    qtcharts
+    qttranslations
```
Comment (sikmir) [inline pkgs/tools/system/stacer/default.nix:23]:
```suggestion
    qttools
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/stacer/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
```
Comment (sikmir) [inline pkgs/tools/system/stacer/default.nix:27]:
```suggestion
  preConfigure = ''
    lrelease stacer/stacer.pro
  '';
 
  postInstall = ''
    install -Dm644 ../translations/*.qm -t $out/share/stacer/translations
  '';

```

File: pkgs/tools/system/stacer/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "stacer";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "oguzhaninan";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0qndzzkbq6abapvwq202kva8j619jdn9977sbqmmfs9zkjz4mbsd";
+  };
+
```
Comment (sikmir) [inline pkgs/tools/system/stacer/default.nix:20]:
```suggestion
  postPatch = ''
    substituteInPlace stacer/Managers/app_manager.cpp \
      --replace 'qApp->applicationDirPath() + "/translations"' \
                'QStandardPaths::locate(QStandardPaths::AppDataLocation, "translations", QStandardPaths::LocateDirectory)'
  '';

```

Comment (sikmir) [reply pkgs/tools/system/stacer/default.nix:20]: I guess, we can submit this fix to upstream.

**Comment by RaitoBezarius** [comment]
> NOTE: Should we include a warning in longDescription that this tool might not work as expected under NixOS and can cause unintended side effects?

Did you test the tool and what kind of side effects did it cause?

**Comment by Zahrun** [comment]
I built stacer from this branch and it built successfully.
I use stacer mostly for the System Cleaner tab.
It successfully cleaned 10Go of Application Logs and Application cache.
It seems Crash Report option does not find the items in /var/lib/systemd/coredump, where I had 5 additional Go of dumps.

@dit7ya, are you around?

**Comment by julius-boettger** [comment]
I just saw that someone requested this package again two weeks ago. I'm also interested.
@dit7ya, are you there?

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 188986` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stacer</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Pull request merged**